### PR TITLE
add throttling for dataplex client

### PR DESCRIPTION
**Pull request by thinhha**

### CODE REVIEW

File: clouddq/integration/dataplex/dataplex_client.py [hunk 1]
```diff
@@ -234,31 +249,7 @@ def delete_dataplex_task(
         )
         return response
 
-    def get_dataplex_iam_permissions(
```
Comment (hejnal) [inline clouddq/integration/dataplex/dataplex_client.py:237]: Is that clean-up intended?

I can still see some references to that function in the clouddq_dataplex.py

def get_iam_permissions(self) -> list:
        body = {"resource": "dataplex", "permissions": ["roles/dataproc.worker"]}
        return self._client.**get_dataplex_iam_permissions**(
            body=body,
        )

Please doublecheck.

File: .github/workflows/build-test.yml [hunk 2]
```diff
@@ -29,7 +29,7 @@ jobs:
     runs-on: ${{ matrix.os }}
     strategy:
       fail-fast: false
-      max-parallel: 2
+      max-parallel: 1
```
Comment (hejnal) [inline .github/workflows/build-test.yml:32]: I guess that would impact the test performance overall (not just dataplex related tasks). Are we keeping track of this, so in the future it can be increased again?

File: scripts/test_pip_install.sh [hunk 3]
```diff
@@ -106,7 +107,7 @@ python3 -m clouddq ALL configs \
 
 if [[ -f "${GOOGLE_APPLICATION_CREDENTIALS:-}" ]]; then
     # test clouddq with exported service account key
-    python3 -m clouddq ALL configs \
+    python3 -m clouddq T1_DQ_1_VALUE_NOT_NULL,T2_DQ_1_EMAIL,T3_DQ_1_EMAIL_DUPLICATE configs \
```
Comment (hejnal) [inline scripts/test_pip_install.sh:110]: Is this change related to the throttling issue? If not, why would you need to explicitly set the bindings to be run?

File: clouddq/integration/dataplex/dataplex_client.py (hunk 1 above)
Comment (thinhha) [reply clouddq/integration/dataplex/dataplex_client.py:237]: oh right thanks! That's not actually used anywhere so I will delete that as well.

File: scripts/test_pip_install.sh (hunk 3 above)
Comment (thinhha) [reply scripts/test_pip_install.sh:110]: Yes because ALL includes rule_bindings with entity_uris which will trigger the dataplex metadata API calls, leading to run failures.

Comment (hejnal) [reply scripts/test_pip_install.sh:110]: ok, gotcha! makes sense

File: .github/workflows/build-test.yml (hunk 2 above)
Comment (hejnal) [reply .github/workflows/build-test.yml:32]: ok, understood!

File: clouddq/integration/dataplex/dataplex_client.py (hunk 1 above)
Comment (hejnal) [reply clouddq/integration/dataplex/dataplex_client.py:237]: perfect, thanks!
